pyintegration/deephaven2/learn/__init__.py

```diff
@@ -0,0 +1,175 @@
+#
+#   Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+#
+""" Deephaven's learn module provides utilities for efficient data transfer between Deephaven tables and Python objects,
+as well as a framework for using popular machine-learning / deep-learning libraries with Deephaven tables.
+"""
+from typing import List, Union, Callable, Type
+
+import jpy
+
+from deephaven2 import DHError
+from deephaven2.table import Table
+
+_JLearnInput = jpy.get_type("io.deephaven.integrations.learn.Input")
+_JLearnOutput = jpy.get_type("io.deephaven.integrations.learn.Output")
+_JLearnComputer = jpy.get_type("io.deephaven.integrations.learn.Computer")
+_JLearnScatterer = jpy.get_type("io.deephaven.integrations.learn.Scatterer")
+
+
+class Input:
+    """ Input specifies how to gather data from a Deephaven table into an object. """
+
+    def __init__(self, col_names: Union[str, List[str]], gather_func: Callable):
+        """  Initializes an Input object with the given arguments.
+
+        Args:
+            col_names (str|list) : column name or list of column names from which to gather input.
+            gather_func (Callable): function that determines how input gets transformed into an object.
+        """
+        self.input = _JLearnInput(col_names, gather_func)
+
+    def __str__(self):
+        """ Returns the Input object as a string. """
+        return self.input.toString()
+
+
+class Output:
+    """ Output specifies how to scatter data from an object into a table column. """
+
+    def __init__(self, col_name: str, scatter_func: Callable, col_type: Type):
+        """ Initializes an Output object with the given arguments.
+
+        Args:
+            col_name (str) : name of the new column that will store results.
+            scatter_func (Callable): function that determines how data is taken from an object and placed into a
+                Deephaven table column.
+            col_type (Type) : desired data type of the new output column, default is None
+        """
+        self.output = _JLearnOutput(col_name, scatter_func, col_type)
+
+    def __str__(self):
+        """ Returns the Output object as a string. """
+        return self.output.toString()
+
+
+def _validate(inputs: Input, outputs: Output, table: Table):
+    """ Ensures that all input columns exist in the table, and that no output column names already exist in the table.
+
+    Args:
+        inputs (Input)  : list of Inputs to validate.
+        outputs (Output) : list of Outputs to validate.
+        table (Table)  : table to check Input and Output columns against.
```

Comment:
I can't fix this with a suggestion, but could these lines be updated such that the colons are vertically aligned?